### PR TITLE
11-incident-type-fact : completed IncidentTypeFactory

### LINKED ISSUE #11: Create IncidentTypeFactory and get method
### Feature Description
Create an `IncidentTypeFactory.cs` inside the `Factories` folder.
Create `GetAll()` method that returns all incident types in the database.

1. Create file
2. Create method

### PR DESCRIPTION
**Pull request by MattKraatz**
## Status
**READY**

## Testing

[X] I have added passing tests for all new functionality.

[X] I have confirmed that all test prior to this pull request still PASS.

## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.

## Description
Generate an Incident Type Factory with the singleton pattern and a method to retrieve all IncidentTypes. Write tests.

## Resolves Issue Number
Resolves #11 

## Deploy Notes

```sh
git pull --prune
git checkout 11-incident-type-fact
dotnet run
```

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

1. Make sure tests pass

## Impacted Files in Application
List general components of the application that this PR will affect:

* Factories/IncidentTypeFactory.cs
* IncidentTypeFactoryTests.cs

### CODE REVIEW

**Comment by matthamil** [comment]
LGTM :+1:

All tests passing on my end.

**Comment by daynewright** [comment]
👍 LGTM! All Tests Pass!

File: customer-service-tests/IncidentTypeFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        [Fact]
+        public void IncidentTypeFactoryIsSingleton()
+        {
+            var fact1 = IncidentTypeFactory.Instance;
+            var fact2 = IncidentTypeFactory.Instance;
+            Assert.Equal(fact1, fact2);
+        }
+
+        [Fact]
+        public void CanGetAllIncidentTypes()
```
Comment (NewEvolution) [inline customer-service-tests/IncidentTypeFactoryTests.cs:29]: Little fuzzy on the name here, you call it "get all" but then only test for the first one in the list.  For a get all, I'd expect something more like where you check the length of the list to equal a length you expect it to have, or check that all the Labels are the labels you expect.

**Review by NewEvolution — COMMENTED** [review]
Looks good overall, one comment on the final test.

**Review by Anulfo — COMMENTED** [review]
LGTM

**Comment by garrett-vangilder** [comment]
All tests pass LGTM!

**Pull request merged**